### PR TITLE
Refactor adapter ibisstore frontend implementation

### PR DESCRIPTION
**Pull request by alisihab**

### CODE REVIEW

File: webapp/src/main/webapp/iaf/gui/js/config.js [hunk 1]
```diff
@@ -101,72 +101,41 @@ angular.module('iaf.beheerconsole').config(['$cookiesProvider', '$locationProvid
 			},
 		},
 	})
-	.state('pages.errorstorage', {
+	.state('pages.storage', {
 		abstract: true,
 		url: "/adapter/:adapter/:receiver/",
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/js/config.js:106]: Could you rename this to `/adapter/:adapter/receiver/:receiver/` the the url is somewhat api compliant?

File: webapp/src/main/webapp/iaf/gui/js/controllers.js [hunk 2]
```diff
@@ -1209,8 +1209,14 @@ angular.module('iaf.beheerconsole')
 	$scope.receiverName = $state.params.receiver;
 	if(!$scope.receiverName)
 		return SweetAlert.Warning("Invalid URL", "No receiver name provided!");
+	$scope.storageType = $state.params.storageType;
+	if(!$scope.storageType)
+		return SweetAlert.Warning("Invalid URL", "No storage type provided!");
 
-	$scope.base_url = "adapters/"+$scope.adapterName+"/receivers/"+$scope.receiverName+"/errorstorage";
+	$state.$current.data.pageTitle = $state.params.storageType;
+	$state.$current.data.breadcrumbs = "Adapter > "+$state.params.storageType;
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/js/controllers.js:1217]: Can you not do this in the config.js like it's done for `pages.storage.view` ?

File: webapp/src/main/webapp/iaf/gui/js/controllers.js [hunk 3]
```diff
@@ -1298,9 +1310,25 @@ angular.module('iaf.beheerconsole')
...
 	}
-
 	$scope.dtOptions = {
 		stateSave: true,
+		stateSaveCallback: function(settings, data) {
+			data.columns = columns;
+			sessionStorage.setItem('DataTable'+$scope.storageType, JSON.stringify(data));
+		},
+		stateLoadCallback: function(settings) {
+			return JSON.parse(sessionStorage.getItem('DataTable'+$scope.storageType));
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/js/controllers.js:1320]: This should be `return filterCookie`;

File: webapp/src/main/webapp/iaf/gui/js/controllers.js [hunk 4]
```diff
@@ -1298,9 +1310,25 @@ angular.module('iaf.beheerconsole')
 			label: true,
 		}
 	}
-
 	$scope.dtOptions = {
 		stateSave: true,
+		stateSaveCallback: function(settings, data) {
+			data.columns = columns;
+			sessionStorage.setItem('DataTable'+$scope.storageType, JSON.stringify(data));
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/js/controllers.js:1317]: This should be `Cookies.set($scope.storageType+"Filter", data);`

File: webapp/src/main/webapp/iaf/gui/views/txstorage/adapter_storage_list.html [hunk 5]
```diff
@@ -30,42 +30,42 @@ <h4>Display Filters</h4>
 							<tbody>
 								<tr>
 									<th>Storage ID</th>
-									<td><input disabled icheck type="checkbox" ng-change="updateFilter('id')" ng-model="displayColumn.id"/></td>
+									<td><input disabled icheck type="checkbox" ng-change="updateFilter('id', storageType)" ng-model="displayColumn.id"/></td>
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/views/txstorage/adapter_storage_list.html:33]: I don't think you need to pass the storageType here, as it's already present as a `$scope.storageType` variable in the controller.

**Review by nielsm5 — CHANGES_REQUESTED** [review]

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**